src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -66,14 +74,116 @@ class EventsResource extends Resource {
+
+            def filters = eventsDAO.getFilters()
+            def fields = eventsDAO.getCustomFields()
+
+            filters.each {
+                csvheader += it.name
+            }
+            fields.each {
+                csvheader += it.name
+            }
+
+            def time = new Date().time
+            String csvfilename = "events_gospel" + time + ".csv"
+            CSVWriter writer = new CSVWriter(new BufferedWriter(new FileWriter(csvfilename)))
+
+            writer.writeNext((String [] ) csvheader.toArray())
+
+            resultObject = getResultObject(events)
+            resultObject.data.each {
+                String eventid = it.id
+                Event baseEvent = it.attributes as Event
+
+                ArrayList<String> baseEventRecord =
+                        [eventid, baseEvent.title, baseEvent.description,
+                         "DATE_FROM_PLACE_HOLDER", "DATE_TO_PLACE_HOLDER",
+                         "START_TIME_PLACE_HOLDER", "END_TIME_PLACE_HOLDER",
+                         baseEvent.location, baseEvent.address, baseEvent.city, baseEvent.state,
+                         baseEvent.eventURL, baseEvent.room, baseEvent.keywords, baseEvent.tags,
+                         baseEvent.photoURL, baseEvent.ticketURL, baseEvent.cost, baseEvent.hashtag,
+                         baseEvent.facebookURL, baseEvent.group, baseEvent.department,
+                         "Allow_User_Activity", "USER_ATTENDANCE_FIELD", baseEvent.visibility,
+                         "FEATURED_TABS", baseEvent.sponsored, baseEvent.venuePageOnly,
+                         baseEvent.excludeFromTrending ]
+
+                //Map over the filter items in the same order as they were added to the header.
+                def filtersMapping = baseEvent.filters.collectEntries { [it.filter, it.items] }
+                filters.each {
+                    def entry = filtersMapping[it.name]
+                    if(entry != null) {
+                        String entryVal = filtersMapping[it.name].toString()
+                        entryVal = entryVal.substring(1 , entryVal.size() - 1) //Trim '[' ']'
+                        System.out.println("Filter Name: " + it.name + " Filter Items: " + entryVal)
+
+                        baseEventRecord.add(entryVal)
+                    } else {
+                        baseEventRecord.add("")
+                    }
+                }
+
+                //Map over the custom field entries in the same order as they were added to the header
+                def eventCFieldsMap = baseEvent.customFields.collectEntries {[it.field,it.value]}
+                fields.each {
+                    String entryVal = eventCFieldsMap[it.name]
+                    if(entryVal == null) {
+                        entryVal = ""
+                    }
+                    baseEventRecord.add(entryVal)
+                }
+
+                List<Instance> eventInstances = eventsDAO.getInstances(eventid)
```

Comment:
Since you used the getResultObject method, you shouldn't have to call the eventsDAO.getInstances method since each ResourceObject in the ResultObject already contains instances.